swarm/storage/mru/cacheentry.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package mru
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+const (
+	hasherCount            = 8
+	resourceHashAlgorithm  = storage.SHA3Hash
+	defaultRetrieveTimeout = 100 * time.Millisecond
```

Comment:
This is very very low. In network scenario this will be longer, so MRU resolver will be fooled

Reply:
This number has been as-is since the first version. What would be a good timeout?.
Take into account that the best-case scenario will take 2-3 lookups.

Reply:
FTR, our convo in orange lounge

lash
@nolash
Sep 17 09:25
maybe the initial lookup should be async and bathced at least, if the timeout increases? And then cancel if the newest comes back?

Javier Peletier
@jpeletier
Sep 17 09:35

Can you elaborate? Each epoch lookup fail/success implies the algorithm takes a diferent path for the next epoch lookup. I thought about this, but would need to run it in parallel, forking on every lookup while we wait for the timeout or success.

So this would allow us to increase the timeout without sacrificing total lookup time, but would produce more lookups.

Viktor Trón
@zelig
Sep 17 09:59
This change is not for this PR for sure. As long as syncing, retrieval works, it's ok. Nonetheless 100millisecs is just too short for network retrieval. I worry that on the main cluster this would result in outdated resolutions cos some chunks just won't arrive that fast.
An alternative to parallelism is to go ahead after a short time but backtrack if a chunk that we thought was not found arrives later. That's probably easier to implement.

Another alternative or additional measure is we have redundancy (probably needed anyway for root chunks) and retrieve alternatives parallelly. That might guarantee some upper limit on successful retrieval latency and therefore makes the 'not found' decision reliable